tests/align/align_create_tensor_ff.py

```diff
@@ -0,0 +1,681 @@
+
+def create_tensors_for_exp_ff(ffmodel):
+  inp: torch.Tensor = gen_tensor(
+      (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+      dtype="float32"
+  )
+  label: torch.Tensor = gen_tensor(
+      (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+      dtype="float32"
+  )
+
+  input_tensor = ffmodel.create_tensor(inp.shape, DataType.DT_FLOAT)
+  output_tensor = ffmodel.exp(
+      x=input_tensor,
+      name="exp"
+  )
+  return ((input_tensor, inp),), label, output_tensor
+    
+def create_tensors_for_flat_ff(ffmodel):
+  INPUT_SIZE_2 = 512
+
+  inp: torch.Tensor = gen_tensor(
+      (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE, INPUT_SIZE_2),
+      dtype="float32"
+  )
+  label: torch.Tensor = gen_tensor(
+      (BATCH_SIZE, INPUT_SIZE * INPUT_SIZE_2 * SEQ_LENGTH),
+      dtype="float32"
+  )
+
+  input_tensor = ffmodel.create_tensor(inp.shape, DataType.DT_FLOAT)
+  output_tensor = ffmodel.flat(
+      input=input_tensor,
+      name="flat"
+  )
+  return ((input_tensor, inp),), label, output_tensor
+ 
+def create_tensors_for_getitem_ff(ffmodel):
+    attention_mask = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH),
+        dtype="float32",
+        low=0,
+        high=2,
+    )
+    label: torch.Tensor = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH),
+        dtype="float32",
+    )  # unused
+
+    attention_mask_tensor = ffmodel.create_tensor(
+        attention_mask.shape,
+        DataType.DT_FLOAT,
+    )
+    extended_attention_mask = GetItemNode.slice_tensor(
+        ffmodel,
+        attention_mask_tensor,
+        (slice(None, None, None), None, None, slice(None, None, None)),
+        "slice",
+    )
+    return ((attention_mask_tensor, attention_mask),), label, extended_attention_mask
+
+def create_tensors_for_identity_ff(ffmodel):
+  inp: torch.Tensor = gen_tensor(
+      (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+      dtype="float32"
+  )
+  label: torch.Tensor = gen_tensor(
+      (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+      dtype="float32"
+  )
+
+  input_tensor = ffmodel.create_tensor(inp.shape, DataType.DT_FLOAT)
+  output_tensor = ffmodel.identity(
+      input=input_tensor,
+      name="identity"
+  )
+  return ((input_tensor, inp),), label, output_tensor
+     
+# def create_tensors_for_layernorm_ff(ffmodel):    
+#     HIDDEN_SIZE = 512
+#     EPS = 1e-6
+#     inp: torch.Tensor = gen_tensor(
+#         (BATCH_SIZE, SEQ_LENGTH, HIDDEN_SIZE),
+#         dtype="float32",
+#     )
+#     label: torch.Tensor = gen_tensor(
+#         (BATCH_SIZE, SEQ_LENGTH, HIDDEN_SIZE),
+#         dtype="float32",
+#     )
+
+#     input_tensor = ffmodel.create_tensor(inp.shape, DataType.DT_FLOAT)
+#     output_tensor = ffmodel.layer_norm(
+#         input=input_tensor,
+#         axes=[len(input_tensor.dims) - 1],  # normalize over the last dimension
+#         elementwise_affine=True,
+#         eps=EPS,
+#         name="layernorm",
+#     )
+#     return ((input_tensor, inp),), label, output_tensor
+
+# def create_tensors_for_linear_ff(ffmodel):
```

Comment:
Why are these commented out? If they are broken, can you create an issue and link it in the response to this comment and in the code by the comment just so we can remember what this is doing here?

Reply:
#625